### ISSUE TITLE
508 Copy Feedback

### ISSUE THREAD

**Issue by nickbristow** [18F/govt-wide-patternlibrary#243]
@juliaelman 
## Inputs

Current `All form control tags should have an associated label. The label’s value for the for attribute should exactly match the value for the input’s id attribute. For example, the input with id="favorite-pie" will always have a label with for="favorite-pie". This way screen readers are able to perceive the relevant content.`

The second sentence is a bit odd maybe `The label's for attribute value should match the related input's id attribute.`

I would also work in something to the effect of `This id value must be unique to the entire page.`

I would also add a section on color contrast. `If the color scheme of the buttons are adjusted ensure all states have a minimum contrast ratio of 4.5:1 (for small text, 3:1 for large) for all states of the button. This includes default, hover, selected, disabled.`

## Form blocks 

When explaining fieldsets, I think mentioning that `a single legend is always required for fieldset` and that you can `embed multiple fieldsets and legends` for more complex forms would be helpful. Also maybe provide the example of `a question with radio button options for answer is a good example of fieldset and legend. The whole group is wrapped in a fieldset with the question itself being the legend`

It might also be worth mentioning that keeping your form blocks in a vertical pattern is ideal for accessibility reason. Basically, limited vision can make it hard to scan from right to left. 

### Reset password

Using aria-describedby is a good way to associate the password strength information with the field

## Search bar 

`Even if the search bar isn't using a label, the form field should include a label for screen reader users`

`The search button itself should be a submit button for the form to reduce the number of keystrokes required to use the form.`

**Comment by maya** [comment]
@nickbristow For color contrast, don't we want this to apply to everything not just inputs?

**Comment by nickbristow** [comment]
@maya yes, color contrast standards apply to all text on a web page. The confusing bit of this happens when people don't realize that it includes all states of a page or element. Like a visited state of a link, or the depressed look of a button. Not like, a sad button, but when you click it and its pressed in.

**Comment by maya** [comment]
@nickbristow so at what points in the library do you think it should be mentioned? Colors, inputs, etc... ?

**Comment by nickbristow** [comment]
@maya I don't think it should be mentioned everywhere, but I do think it should be mentioned for the buttons, because people miss it.

**Comment by maya** [comment]
:+1:

**Comment by carodew** [comment]
Is now the time that I tell @nickbristow that I keep a record of funny things people say at every job I've ever been to? Congrats, you've been added to the archives. 

" Like a visited state of a link, or the depressed look of a button. Not like, a sad button, but when you click it and its pressed in."

**Comment by nickbristow** [comment]
@carodew 😬
